src/nvim/ui.c

```diff
@@ -147,6 +148,18 @@ bool ui_rgb_attached(void)
   return false;
 }
 
+/// Returns true if a GUI is attached.
+bool ui_gui_attached(void)
+{
+  for (size_t i = 0; i < ui_count; i++) {
+    bool tui = uis[i]->stdin_tty && uis[i]->stdout_tty;
```

Comment:
any UI with `!stdin_tty` or `!stdout_tty` is considered a GUI

Reply:
hmm I would change `&&` to `||` here. `cat file | nvim -` will still use the TUI and only have stdout_tty set